File: app/src/main/java/com/laushkina/anastasia/psychosophy/view/relationships/PsychotypesAdapter.java
package com.laushkina.anastasia.psychosophy.view.relationships;

import android.content.Context;
import android.graphics.Color;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.laushkina.anastasia.psychosophy.domain.Psychotype;
import com.laushkina.anastasia.psychosophy.view.psychotypeDescription.PsychotypeDescriptionGetter;

public class PsychotypesAdapter extends ArrayAdapter<Psychotype> {

    private int resourceLayout;
    private int firstItemColor;
    private String promtTest;

    PsychotypesAdapter(@NonNull Context context, int resource, Psychotype[] psychotypes, String promtTest, int firstItemColor) {
        super(context, resource, psychotypes);
        this.resourceLayout = resource;
        this.promtTest = promtTest;
        this.firstItemColor = firstItemColor;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            convertView = LayoutInflater.from(getContext()).inflate(resourceLayout, null);
        }

        initText(convertView, position);
        return convertView;
    }

    @Override
    public View getDropDownView(int position, View convertView, ViewGroup parent) {
        if(convertView == null){
            convertView = LayoutInflater.from(getContext()).inflate(resourceLayout, null);
        }

        initText(convertView, position);
        return convertView;
    }

    private void initText(View convertView, int position){
        TextView textView = convertView.findViewById(android.R.id.text1);
        textView.setText(getText(position));
        if (position == 0) {
            textView.setTextColor(firstItemColor);
        } else  {
            textView.setTextColor(Color.BLACK);
        }
    }

    private CharSequence getText(int position){
        Psychotype psychotype = getItem(position);
        //First item is prompt
        return position == 0 ? promtTest : PsychotypeDescriptionGetter.getTitle(psychotype, getContext());
    }

}

File: app/src/main/java/com/laushkina/anastasia/psychosophy/view/relationships/PsychotypesAdapter.kt
package com.laushkina.anastasia.psychosophy.view.relationships

import android.content.Context
import android.graphics.Color
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import com.laushkina.anastasia.psychosophy.domain.Psychotype
import com.laushkina.anastasia.psychosophy.view.psychotypeDescription.PsychotypeDescriptionGetter

class PsychotypesAdapter(context: Context,
                         psychotypes: Array<Psychotype?>,
                         private val resourceLayout: Int,
                         private val promtTest: String,
                         private val firstItemColor: Int): ArrayAdapter<Psychotype?>(context, resourceLayout, psychotypes) {

    override fun getView(position: Int, view: View?, parent: ViewGroup): View {
        val convertView = view ?: LayoutInflater.from(context).inflate(resourceLayout, null)

        initText(convertView, position)
        return convertView
    }

    override fun getDropDownView(position: Int, view: View?, parent: ViewGroup): View? {
        return getView(position, view, parent)
    }

    private fun initText(convertView: View, position: Int) {
        val textView = convertView.findViewById<TextView>(android.R.id.text1)
        textView.text = getText(position)
        if (position == 0) {
            textView.setTextColor(firstItemColor)
        } else {
            textView.setTextColor(Color.BLACK)
        }
    }

    private fun getText(position: Int): CharSequence {
        val psychotype = getItem(position)
        //First item is prompt
        return if (position == 0) promtTest else PsychotypeDescriptionGetter.getTitle(psychotype, context)
    }
}